src/SqlAsyncCollector.cs

```diff
@@ -95,7 +96,14 @@ public async Task AddAsync(T item, CancellationToken cancellationToken = default
                 TelemetryInstance.TrackEvent(TelemetryEventName.AddAsync);
                 try
                 {
-                    this._rows.Add(item);
+                    if (typeof(T) == typeof(string))
+                    {
+                        this._rows.Add(JsonConvert.DeserializeObject<T>(item as string));
```

Comment:
This seems odd to me too. item is a string, and then we're calling JsonConvert.DeserializeObject on it...but T is also string. So we're deserializing a string into...a string? I'm pretty sure this is also doing nothing unless there's some behavior I'm missing.

Reply:
not really sure, but that's the same thing cosmosdb folks have on their repo, I just replicated what they suggested and this code is currently not being hit since what we actually get is a JObject.

Reply:
We shouldn't be doing something just because their code is doing it - we need to know what our code is doing and how it works regardless of what their implementation looks like.

I'd suggest removing this if the code isn't even being hit. If that causes problems then we can dive into what exactly needs to happen here.

Reply:
Well, those are changes they suggested to add javascript support, I wasn't really sure what works and what don't until the entire thing came along. And it made sense that when we get the input from javascript as a string, to deserialize it and add it to rows.